### PR TITLE
Feat/player class

### LINKED ISSUE #13: Player Class
Player

Represents one player.

Fields

String name;
double balance;
ImageIcon icon;
Set<PropertyTile> properties;
boolean inJail;
int jailTurnCount;
int position;
boolean active;

Methods

buy(double amount)
sell(double amount)
canAfford(double amount)
addProperty(PropertyTile property)
removeProperty(PropertyTile property)
sellProperty(PropertyTile property)
goToJail(int jailPosition)
leaveJail()
isBankrupt()

Description
Stores player money, location, owned properties, and jail state.

### PR DESCRIPTION
**Pull request by OmoT7**
This pull request introduces a new `Player` class and a minimal `Properties` class to the `model` package, and adds comprehensive Boundary Value Analysis (BVA) documentation for the `Player` class. It also removes the generic BVA instructions from the `README.md` and moves them, along with new, concrete test cases, into `player.md`. The most important changes are summarized below:

### Model Implementation

* Added a new `Player` class in `model/Player.java` that implements player state and behavior, including balance management, property ownership, jail mechanics, and bankruptcy logic. The class includes methods such as `buy`, `sell`, `canAfford`, `addProperty`, `removeProperty`, `sellProperty`, `goToJail`, `leaveJail`, and `isBankrupt`.
* Added a minimal `Properties` class in `model/Properties.java` with a stub `getPrice` method, to support property-related operations in the `Player` class.

### Documentation Updates

* Removed the general BVA instructions and example formats from `docs/bva/README.md`, streamlining the documentation to avoid duplication.
* Created a new `docs/bva/player.md` file that includes both the BVA instructions and a detailed, method-by-method BVA analysis for the `Player` class, listing concrete test cases for each public method.

### CODE REVIEW

**Review by Andy2887 — APPROVED** [review]

**Review by ChefRedr — CHANGES_REQUESTED** [review]
For `buy(double amount)`, your BVA specifies 6 tests. Remove the unnecessary tests for `amount = 50` and `Double.MIN_VALUE`, since they are not in the BVA.

Also you're missing test TC4: Buy with slightly less than balance

File: src/test/java/model/PlayerTests.java [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+
+        assertTrue(success, "Buying with exact funds should be successful");
+        assertEquals(0, player.getBalance(), 0.001, "Balance should = 0 after buying with exact funds");
+    }
+
+    @Test
+
+    public void Tests_Buying_With_Slightly_Less_Than_Balance() {
+        Player player = new Player("John", 100.0);
+        boolean success = player.buy(99.0);
```
Comment (ChefRedr) [inline src/test/java/model/PlayerTests.java:57]: Your BVA specifies "State of the system: balance = 100, amount = 99.99" but here you are only buying 99. Test cases should match the BVA documentation.

File: src/test/java/model/PlayerTests.java [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+    @Test
+    public void Tests_Selling_With_Valid_Money() {
+        Player player = new Player("John", 100.0);
+
+        // Passed value is 50.0. Added amount should be 50.0 * 0.8 = 40.0.
+        // New balance = 100.0 + 40.0 = 140.0
+        boolean success = player.sell(50.0);
+
+        assertTrue(success, "Selling with valid amount should be successful");
+        assertEquals(140.0, player.getBalance(), 0.001, "Balance should increase by 80% of the sale amount");
```
Comment (ChefRedr) [inline src/test/java/model/PlayerTests.java:103]: I think this test is correct because you specified sell() to be: this.balance += price * 0.8; However, your BVA for this test specifies that when balance = 100, amount = 50, then the expected should be 150. Modify the BVA to match

File: src/main/java/model/Player.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            return false;
+        }
+        return this.balance >= price;
+    }
+
+    public Set<Properties> getOwnedProperties() {
+        return this.ownedProperties;
+    }
+
+    public boolean addProperty(Properties proptery){
```
Comment (ChefRedr) [inline src/main/java/model/Player.java]: Typo, you misspelled proptery as proptery.

File: src/main/java/model/Properties.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package model;
+
+public class Properties {
```
Comment (ChefRedr) [inline src/main/java/model/Properties.java]: I don't recall having a Properties class in our initial design, can you add that to the doc. Also if this class represents one property, I would consider naming it Property instead of Properties

File: src/test/java/model/PlayerTests.java [hunk 5]
```diff
@@ -0,0 +1,413 @@
...
+    // ==================================================================================================
+    @Test
+    public void Test_GoToJail_Valid_Position() {
+        Player player = new Player("John", 100.0);
+        // Note: based on your Player class logic, valid bounds are 0 to 31.
+        boolean success = player.goToJail(8);
+        
+        assertTrue(success, "Sending player to valid jail position should be successful");
+        assertTrue(player.inJail(), "Player should be marked as in jail");
+        assertEquals(10, player.getPosition(), "Player position should be set to 10 (the jail tile)");
```
Comment (ChefRedr) [inline src/test/java/model/PlayerTests.java:319]: Earlier in this test case you wrote "boolean success = player.goToJail(8);" but then in your assertion you said "assertEquals(10, player.getPosition(), "Player position should be set to 10 (the jail tile)");" Is there an inconsistency here?

File: src/test/java/model/PlayerTests.java (hunk 1 above)
Comment (ChefRedr) [inline src/test/java/model/PlayerTests.java:57]: Your BVA specifies "State of the system: balance = 100, amount = 99.99" but here you are only buying 99. Test cases should match the BVA documentation.

File: src/test/java/model/PlayerTests.java (hunk 2 above)
Comment (ChefRedr) [inline src/test/java/model/PlayerTests.java:103]: I think this test is correct because you specified sell() to be: this.balance += price * 0.8; However, your BVA for this test specifies that when balance = 100, amount = 50, then the expected should be 150. Modify the BVA to match

**Review by ChefRedr — CHANGES_REQUESTED** [review]
Left some comments, I think after these are solved then we should be able to merge.

File: src/main/java/model/Properties.java (hunk 4 above)
Comment (OmoT7) [reply src/main/java/model/Properties.java]: Mb coding late at night does that to an individual

File: src/test/java/model/PlayerTests.java [hunk 6]
```diff
@@ -0,0 +1,414 @@
...
+        EasyMock.replay(propertyMock);
+
+        boolean success = player.sellProperty(propertyMock);
+
+        assertFalse(success, "Selling an unowned property should fail");
+        assertEquals(100.0, player.getBalance(), 0.001, "Balance should remain unchanged");
+    }
+
+    @Test
+    public void Test_Selling_Null_Property() {
```
Comment (ChefRedr) [inline src/test/java/model/PlayerTests.java:299]: Not documented in BVA

File: src/test/java/model/PlayerTests.java [hunk 7]
```diff
@@ -0,0 +1,414 @@
...
+
+        EasyMock.replay(propertyMock);
+
+        boolean success = player.removeProperty(propertyMock);
+
+        assertFalse(success, "Removing a property from a player with no properties should be rejected");
+    }
+
+    @Test
+    public void Test_Removing_Null_Property_From_Player() {
```
Comment (ChefRedr) [inline src/test/java/model/PlayerTests.java:256]: Not documented in BVA

**Review by ChefRedr — CHANGES_REQUESTED** [review]
2 Extra Tests that aren't documented in BVA

**Review by ChefRedr — APPROVED** [review]
Nicely done
